src/group_theory/submonoid/operations.lean

```diff
@@ -525,6 +525,21 @@ assume p hp, prod.fst_mul_snd p ▸ mul_mem _
   ((le_sup_left : s.prod ⊥ ≤ s.prod ⊥ ⊔ prod ⊥ t) ⟨hp.1, set.mem_singleton 1⟩)
   ((le_sup_right : prod ⊥ t ≤ s.prod ⊥ ⊔ prod ⊥ t) ⟨set.mem_singleton 1, hp.2⟩)
 
+@[to_additive]
+lemma mem_map_equiv {f : M ≃* N} {K : submonoid M} {x : N} :
+  x ∈ K.map f.to_monoid_hom ↔ f.symm x ∈ K :=
+@set.mem_image_equiv _ _ ↑K f.to_equiv x
+
+@[to_additive]
+lemma map_equiv_eq_comap_symm (f : M ≃* N) (K : submonoid M) :
+  K.map f.to_monoid_hom = K.comap f.symm.to_monoid_hom :=
+by ext; simp only [mem_map_equiv]; simp
+
+@[to_additive]
+lemma comap_equiv_eq_map_symm (f : N ≃* M) (K : submonoid M) :
+  K.comap f.to_monoid_hom = K.map f.symm.to_monoid_hom :=
+by rw [map_equiv_eq_comap_symm, mul_equiv.symm_symm]
```

Comment:
```suggestion
(map_equiv_eq_comap_symm f.symm K).symm
```

Reply:
etc for all the other changes in this patch.

Reply:
I made these changes, but I'm not really convinced it's an improvement. It relies on a definition equality `f.symm.symm.to_monoid_hom = f.to_monoid_hom` which is slightly bad practice. It's less readable and only about 10ms faster.

Reply:
Thanks - I suppose for this one you could argue this is defeq abuse; but for the `coe_injective` version, this definitional equality is part of the API, and the `simp` proof were presumably much slower.

If you feel strongly, feel free to revert back to the `rw` proofs for `comap_equiv_eq_map_symm `

Reply:
I don't feel strongly. Getting rid of the `simp` is fair enough, but personally I don't think this sort of change is worth the time spent on it.